packages/core/src/tracing/instrumentation/protocols/httpClient.js

```diff
@@ -19,17 +19,24 @@ const {
 const constants = require('../../constants');
 const cls = require('../../cls');
 const url = require('url');
+const requireHook = require('../../../util/requireHook');
+let logger;
+logger = require('../../../logger').getLogger('tracing/httpClient', newLogger => {
+  logger = newLogger;
+});
 
 let extraHttpHeadersToCapture;
 let isActive = false;
 
 exports.init = function init(config) {
   instrument(coreHttpModule, false);
   instrument(coreHttpsModule, true);
-
   extraHttpHeadersToCapture = config.tracing.http.extraHttpHeadersToCapture;
+  requireHook.onModuleLoad('request', logDeprecatedWarning);
 };
-
+function logDeprecatedWarning() {
+  logger.warn('The support for request has been deprecated.');
```

Comment:
Suggestion: One unified message format for deprecation.

> logger.warn('kafka-node is deprecated. The support will be removed in the next major release.');
> logger.warn('The support for request has been deprecated.');
>   logger.warn(
    'The support for @apollo-federation has been deprecated, following its official end-of-life in September 2023.'
  );

Reply:
Would it be possible to consider incorporating the following log messages for deprecation warnings?

[Deprecation Warning]: The support for request has been deprecated.
[Deprecation Warning]: The support for @apollo-federation has been deprecated.

Reply:
Two more things:
- maybe we can add the link to the target npm package?
- What about "The support will be removed in the next major release"?

Reply:
>  logger.warn('[Deprecation Warning] The support for "request" is deprecated and will be removed in the next major release. Please consider migrating to an appropriate package.');
> 
> logger.warn('[Deprecation Warning] The support for "@apollo-federation" is deprecated and will be removed in the next major release. Please consider migrating to an appropriate package.');
> 
> logger.warn('[Deprecation Warning] The "kafka-node" library is deprecated and will be removed in the next major release. Please consider migrating to an appropriate package.');

could you please check these updates